### PR TITLE
added error handling. Updated magick convert command

### PR DESCRIPTION
**Pull request by kem-a**

### CODE REVIEW

File: AppImage-thumbnailer [hunk 1]
```diff
@@ -4,19 +4,34 @@ outFile="$(realpath -m "$2")"
...
+
+if [ ! -f "$thumb" ]; then
+  cd .. && rm -r "$tmpDir"
+  exit 1
+fi
+
+if ! magick "$thumb" -background none -thumbnail "$size" "../$tmpThumb"; then
+  cd .. && rm -r "$tmpDir"
+  exit 1
+fi
```
Comment (realmazharhussain) [inline AppImage-thumbnailer:33]: These lines add complexity for the sake of it.

In a real setting, `thumb` is pretty much guaranteed to exist, and `magick` command is pretty much guaranteed to never fail.

If we are worried about unexpected behavior in unordinarily cases, we can use `#!/usr/bin/bash -e` shebang along with a `rm -rf "$tmpDir"` `EXIT` `trap`.

**Review by realmazharhussain — CHANGES_REQUESTED** [review]
Thank you very much for taking interest in this project. I have requested some changes.

> [!NOTE]
> You might wanna switch to a better AppImage thumbnailer. This thumbnailer depends on the AppImage file being executable, and it also executes the AppImage file in order to fetch its thumbnail which is a huge security/privacy risk.

**Comment by kem-a** [comment]
Unfortunately there are no working AppImage thumbnailer that works, at least none that I've tried.

Regarding security risks I can not comment a lot, but AppImage is set executable by the user and runs in a container, so I assume it's fine. There are not many options to access content inside AppImage without extracting entire content and creating unnecessary writes to the disk and CPU overhead.

edit: but I'll keep looking, before I'm bored

File: AppImage-thumbnailer (hunk 1 above)
Comment (kem-a) [reply AppImage-thumbnailer:33]: Sure, maybe it is a little overkill.

**Comment by realmazharhussain** [comment]
> AppImage is set executable by the user and runs in a container

Well, that is true.

<hr>

There is also https://github.com/linuxmint/xapp-thumbnailers, and it seems to be doing things right.

**Comment by kem-a** [comment]
> > AppImage is set executable by the user and runs in a container
> 
> Well, that is true.
> 
> There is also https://github.com/linuxmint/xapp-thumbnailers, and it seems to be doing things right.

It generates icons for some AppImages, but not all, in my case for less than half. I didn't go down the rabbit hole to figure it out, but people should be put in prison for not listing dependencies for their projects.

Anyway, I created my own project using 7z, also it does not require app to be executable. Can probably drop my PR.

**Comment by realmazharhussain** [comment]
Nice.

**Pull request closed without merging**